service/src/main/java/tds/session/services/impl/SessionServiceImpl.java

```diff
@@ -23,4 +23,11 @@ public SessionServiceImpl(SessionRepository sessionRepository) {
     public Optional<Session> getSessionById(UUID id) {
         return sessionRepository.getSessionById(id);
     }
+
+    @Override
+    public void pause(final UUID sessionId, final String newStatus) {
+        // TODO:  Throw an exception if attempting to pause a Session that is already paused?
```

Comment:
At the risk of getting off-topic: You bring up another interesting point; after a `Session` is paused, we need to go through and pause all the `Exam`s that are members of that `Session`.  Should we assume it is the caller's responsibility to orchestrate that?  For example, if the Proctor pauses a `Session` (for whatever reason) it'll call the `Session` service to pause the session.  The Proctor will _also_ have to call the `Exam` service to pause a collection of exams.